src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -378,7 +362,7 @@ internal sealed class AsyncForwardEntryPoint : SynthesizedEntryPointSymbol
 
             public override TypeSymbol ReturnType => _getAwaiterGetResultCall.Type;
 
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
```

Comment:
The reason that `AsyncForwardEntryPoint` makes the call in the constructor is because we need the diagnostics that come out of that method even if we aren't emitting. (`CreateBody` is only called during emit).

However, when it comes to scripting, (my understanding is that) there's not a huge benefit to getting those diagnostics earlier, so I put all of the synthesis in the `CreateBody` method where it should be.  

I could easily put the `GetAwaitableExpressionInfo` in the constructor, but I don't see a huge reason to.